server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -455,6 +457,19 @@ public T get(Settings settings) {
         return get(settings, true);
     }
 
+    /**
+     * Returns the settings value. If the setting is not present in the given settings object NULL is returned instead.
+     */
+    @Nullable
+    public T getOrNull(Settings settings) {
```

Comment:
Couldn't we change the `COMPRESSION_SETTING` definition to have `null` as default value instead of an empty string, or do null values result in an error elsewhere?

Reply:
Unfortunately the `Setting` implementation do not support `null` default values at various places. We could of course try to adjust this, but could be a bigger thing.